smoke-tests/src/test/java/com/splunk/opentelemetry/HecTelemetryRetriever.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.model.ClearType;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+
+class HecTelemetryRetriever {
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private static final JsonFactory JSON_FACTORY = new JsonFactory(OBJECT_MAPPER);
+  private static final String HEC_PATH = "/services/collector/event";
+
+  private final MockServerClient client;
+
+  HecTelemetryRetriever(int backendPort) {
+    this.client = new MockServerClient("localhost", backendPort);
+  }
+
+  void initializeEndpoints() {
+    client
+        .when(HttpRequest.request(HEC_PATH))
+        .respond(HttpResponse.response("").withStatusCode(200));
+  }
+
+  void clearTelemetry() {
+    client.clear(HttpRequest.request(), ClearType.LOG);
+  }
+
+  List<JsonNode> waitForEntries() throws IOException, InterruptedException {
+    int previousSize = 0;
+    long deadline = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(30);
+    HttpRequest[] requests = new HttpRequest[0];
+
+    while (System.currentTimeMillis() < deadline) {
+      requests = client.retrieveRecordedRequests(HttpRequest.request(HEC_PATH));
+
+      if (requests.length > 0 && requests.length == previousSize) {
```

Comment:
I don't get this. Why does two consecutive responses with the same length indicate doneness? This seems fragile.

Reply:
This is pretty much copy-paste logic from the Otel telemetry retriever, so if it had worked fine there figured it should be fine. Though ideally both should do the "retry until response passes verification" approach instead, but that would require larger changes.